projects/plugins/boost/app/modules/critical-css/class-critical-css.php

```diff
@@ -663,14 +663,9 @@ public function asynchronize_stylesheets( $html, $handle, $href, $media ) {
 			return $html;
 		}
 
-		// If the stylesheet is not meant for screen, do not alter the stylesheet loading.
-		if ( ! in_array( $media, array( 'all', 'screen' ), true ) ) {
-			return $html;
-		}
-
 		$available_methods = array(
-			'async'    => 'media="not all" onload="this.media=\'all\'"',
-			'deferred' => 'media="not all"',
+			'async'    => 'media="not all" data-media="' . $media . '" onload="this.media=\'' . $media . '\'; delete this.dataset.media; this.removeAttribute( \'onload\' );"',
```

Comment:
Because we're already storing the media in` data-media`, I think it'd be better to use that when flipping on the stylesheet instead of rendering JS variables with PHP, which can be a security concern.

Actually, speaking of security - maybe we should `esc_attr( $media )` just to be safe? Or is that completely unnecessary here?

In any case, I think this part:

```php
onload="this.media=\'' . $media . '\';
```

Should be something like this:


```html
onload="this.media=this.dataset.media
```

Also. I wonder if we can save on characters if we make it an inline function that can be reused in all `onload="` calls